pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -117,6 +117,13 @@ def __getstate__(self):
         state["predict_progress_bar"] = None
         return state
 
+    @staticmethod
+    def _resolve_refresh_rate(refresh_rate: Optional[int]) -> int:
+        if os.getenv("COLAB_GPU") and refresh_rate in [None, 1]:
+            # smaller refresh rate on colab causes crashes, choose a higher value
+            refresh_rate = 20
```

Comment:
should this be set if `refresh_rate < 20` ? should this override take effect even if users explicitly set something lower? since the framework is setting a default, it should include a debug log statement as well for observability

Reply:
> should this be set if refresh_rate < 20 ?

Could be a follow up PR. As it is a refactor PR. I will add a log statement here